src/main/scala/rise/core/types/Constraints.scala

```diff
@@ -44,8 +44,11 @@ case class NatCollectionConstraint(a: NatCollection, b: NatCollection)
 }
 
 object Constraint {
-  val canBeSubstituted : Set[Kind.Identifier] => Kind.Identifier with Kind.Explicitness => Boolean =
-    preserve => i => !(preserve.contains(i) || i.isExplicit)
+  def canBeSubstituted(preserve: Set[Kind.Identifier],
```

Comment:
Why this change in style?

Reply:
This allows function overloading for both `i: Kind.Identifier with Kind.Explicitness` and `i: TypeIdentifier`.